### PR TITLE
Update to NAN 2

### PR DESCRIPTION
**Pull request by kkoopa**

### CODE REVIEW

**Comment by ruimarinho** [comment]
Working fine on iojs v3.0.0 (OSX, El Cap). Thanks!

**Comment by kkoopa** [comment]
Note that I had to disable one test and tweak another. Don't know what the proper fixes are.

**Comment by brianc** [comment]
Dang I really appreciate the effort you put into this.  I don't have time to totally dive into it yet & can't merge it until we figure out how to get those tests back to a passing status, but it's on my radar to get working completely. This is a huge amount of work & it's greatly appreciated...i'll try to start in on it this weekend.

**Comment by fundon** [comment]
@kkoopa :+1: 

@brianc Nice!

**Comment by fundon** [comment]
@brianc Could you merge this, please?

**Comment by brianc** [comment]
@fundon I can't merge this until all the tests are passing - I haven't had time to fix it.

**Comment by Qard** [comment]
With node 4.0.0 about to drop, it'd be good to get this working soon.

**Comment by bkw** [comment]
Regarding the [commented-out test](https://github.com/brianc/node-libpq/pull/25/files#diff-7e618394b89a643961240ce520c9ed39R23) : I commented it in and it passes.

iojs 3.3.0, OSX 10.10.5, postgres 9.4.4

File: src/connection.cc [hunk 1]
```diff
@@ -666,68 +553,59 @@ NAN_METHOD(Connection::Notifies) {
...
   //optional error message
 
-  bool sendErrorMessage = args.Length() > 0;
-  char* msg = NULL;
+  bool sendErrorMessage = info.Length() > 0;
+  Nan::Utf8String msg(info[0]);
   if(sendErrorMessage) {
-    msg = NewCString(args[0]);
-    TRACEF("Connection::PutCopyEnd:%s\n", msg);
+    TRACEF("Connection::PutCopyEnd:%s\n", *msg);
```
Comment (bkw) [inline src/connection.cc]: In the non-cancel case *msg is still not null, that's why the copy-in tests fail.
Passing in a char* msg = NULL makes them succeed.

File: src/connection.cc [hunk 2]
```diff
@@ -666,68 +553,59 @@ NAN_METHOD(Connection::Notifies) {
...
   }
 
-  int result = PQputCopyEnd(self->pq, msg);
+  int result = PQputCopyEnd(self->pq, *msg);
```
Comment (bkw) [inline src/connection.cc]: In the non-cancel case msg is still not null, that's why the copy-in tests fail.
Passing in a char msg = NULL makes them succeed.

**Comment by bkw** [comment]
I created a branch on top of @kkoopa 's that puts the tests back to normal and fixes the copy-in behaviour. Probably somebody should clean up bkw/node-libpq@eb6c6ff03 but you'll get the idea.
Thanks to @kkoopa for all the hard work!

**Pull request merged**